lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -515,4 +533,24 @@ sub _for_every_batch {
     return @ret;
 }
 
+# Based on https://www.suse.com/c/multi-factor-authentication-on-suses-build-service/
+sub _bs_ssh_sign ($key, $realm, $value) {
+    die "SSH Key File not found at $key" unless (-e $key || -z $key);
```

Comment:
I don't think the `|| -z $key` part makes sense. This is basically `if !(-e $key || -z $key)` which is `if !-e $key && !-z $key` but you likely want to `die` *if* the file is zero-sized.

I would simplify the condition:

```suggestion
    die "SSH key file not found at $key" unless -s $key;
```

I would generally avoid `unless` with any complex expressions.

Reply:
That's why I avoid more than one condition in an `unless`.

Reply:
I would also go with `unless -s`.
And the `unless (-e $key || -z $key)` is wrong, I double checked, I just don't know why the tests simulating a missing / empty file are passing then...